### PR TITLE
doc: improve note on choosing posix mingw32

### PR DESCRIPTION
**Pull request by jarolrod**
The current [windows build doc](https://github.com/bitcoin/bitcoin/blob/master/doc/build-windows.md) can lead someone to believe that the step where you must choose the posix mingw32 g++ compiler option is only for `Ubuntu 18.04`. It is only until you (or just me) go through the cross-compile process and realize that it's not building because you didn't set this option on > `Ubuntu 18.04`. Then you come back and read the footnotes and see: `Starting from Ubuntu Xenial 16.04...`.

This PR improves this portion of the doc by editing the text around this stage to state "this is now the next step". We could add a note saying `Ubuntu 18.04 and up`, but this is redundant as it's unlikely someone will be using < Ubuntu 18.04 since it's not officially supported by our build system. While here, some minor fixups and add some more guidance to completing this step.

**Master:** [render](https://github.com/bitcoin/bitcoin/blob/master/doc/build-windows.md#building-for-64-bit-windows)
**PR:** [render](https://github.com/bitcoin/bitcoin/blob/dafab2b1b37d1966610b2189e71c52e3af38dfaa/doc/build-windows.md#building-for-64-bit-windows)

### CODE REVIEW

**Comment by dongcarl** [comment]
Huh... Perhaps our `configure.ac` should also search for `x86_64-w64-mingw32-g++-posix` first before `x86_64-w64-mingw32-g++`? Just an idea.

File: doc/build-windows.md [hunk 1]
```diff
@@ -81,9 +81,26 @@ The first step is to install the mingw-w64 cross-compilation tool chain:
 
     sudo apt install g++-mingw-w64-x86-64
 
-Ubuntu Bionic 18.04 <sup>[1](#footnote1)</sup>:
+Next, set the default `mingw32 g++` compiler option to POSIX (For Ubuntu Bionic 18.04 and up<sup>[1](#footnote1)</sup>):
```
Comment (hebasto) [inline doc/build-windows.md:84]: Why mention "18.04 and up"? If this is Ubuntu-specific, I don't think anyone use pre-xenial distros.

Comment (jarolrod) [reply doc/build-windows.md:84]: true ... I guess that part can be dropped, and the note is improved by just saying "this is the next step"

**Comment by jarolrod** [comment]
updated from 6e745eb -> dafab2b ([pr22088.01](https://github.com/jarolrod/bitcoin/tree/pr22088.01) -> [pr22088.02](https://github.com/jarolrod/bitcoin/tree/pr22088.02), [diff](https://github.com/jarolrod/bitcoin/compare/pr22088.01...jarolrod:pr22088.02))

**Changes:**
- Address @hebasto [comment](https://github.com/bitcoin/bitcoin/pull/22088#discussion_r640915804).
  -  Drop specifying "Ubuntu 18.04 and up" in favor of just saying "This is the next step". This is ok to do as it's unlikely someone will use < Ubuntu 18.04 since it's not officially supported by our build system.

File: doc/build-windows.md (hunk 1 above)
Comment (jarolrod) [reply doc/build-windows.md]: addressed in dafab2b

**Review by hebasto — APPROVED** [review]
ACK dafab2b1b37d1966610b2189e71c52e3af38dfaa

Also, it seems reasonable to s/"both the 32 and 64 bit Mingw-w64 packages install"/"Mingw-w64 package installs"/ in the footnote, don't it?

**Comment by laanwj** [comment]
Clearly an improvement, agree that it was confusing
ACK dafab2b1b37d1966610b2189e71c52e3af38dfaa

**Pull request merged**